specs/bellatrix/fork-choice.md

```diff
@@ -171,7 +171,7 @@ def on_block(store: Store, signed_block: SignedBeaconBlock) -> None:
     state_transition(state, signed_block, True)
 
     # [New in Bellatrix]
-    if is_merge_transition_block(pre_state, block.body):
+    if compute_epoch_at_slot(pre_state.slot) >= BELLATRIX_FORK_EPOCH and is_merge_transition_block(pre_state, block.body):
```

Comment:
In Lodestar we have a type guard before that assertion, so +1 on silently handling it

```ts
if (bellatrix.isBellatrixStateType(state as BeaconStateBellatrix) &&
    bellatrix.isBellatrixBlockBodyType(block.body) &&
    bellatrix.isMergeTransitionBlock(state as BeaconStateBellatrix, block.body))
```


https://github.com/ChainSafe/lodestar/blob/e12bcd15f110c99cba73b63a367aea046d3b5bf0/packages/fork-choice/src/forkChoice/forkChoice.ts#L334

Reply:
This check skips transition block validation if the first block in the Bellatrix epoch is the merge transition block. It should rather be placed in `is_merge_transition_complete` function to make it return `False` if `compute_epoch_at_slot(state.slot) < BELLATRIX_FORK_EPOCH`

Reply:
Makes sense!